File: boj/boj-1059.py
import sys
from itertools import combinations
input = sys.stdin.readline

l = int(input())
array = list(map(int, input().split()))
array.sort()
n = int(input())

if n in array:
    print(0)
else:
    start_val, end_val = 0, 0
    for idx, num in enumerate(array):
        if num > n:
            start_val = array[idx-1] + 1
            if idx == 0:
                start_val = 1
            end_val = array[idx] - 1
            break
    
    two_nums_list = list(combinations(range(start_val, end_val+1), 2))

    result = 0
    for (low, high) in two_nums_list:
        if low <= n <= high:
            result += 1
    
    print(result)

File: boj/swift/boj-1059.swift
import Foundation

let l = Int(readLine()!)!
var array: [Int] = readLine()!.split(separator: " ").map { Int(String($0))! }
array.sort()
let n = Int(readLine()!)!

if array.contains(n) {
    print(0)
} else {
    var left = 0, right = 0
    for (idx, num) in array.enumerated() {
        if n < num {
            right = array[idx]
            break
        }
        else {
            left = array[idx]
        }
    }
    right -= 1
    left += 1

    print((right - left) + ((n-left) * (right-n)))
}